tests/test_torch_util.py

```diff
@@ -0,0 +1,303 @@
+"""
+Test :mod:`returnn.torch.util`.
+"""
+
+from __future__ import annotations
+
+import _setup_test_env  # noqa
+
+import os
+import sys
+import unittest
+import torch
+
+from returnn.util import better_exchook
+from returnn.torch.util.gradient_checkpoint import gradient_checkpoint_scope
+
+
+def test_gradient_checkpoint_scope():
```

Comment:
I like this test. Let's see how stable/deterministic it turns out.

Reply:
Yea...